go-controller/pkg/ovn/base_network_controller_pods.go

```diff
@@ -72,19 +78,41 @@ func (bnc *BaseNetworkController) allocatePodIPs(pod *kapi.Pod,
 }
 
 func (bnc *BaseNetworkController) deleteStaleLogicalSwitchPorts(expectedLogicalPorts map[string]bool) error {
-	// get all the nodes from the watchFactory
-	nodes, err := bnc.watchFactory.GetNodes()
-	if err != nil {
-		return fmt.Errorf("failed to get nodes: %v", err)
+	var switchNames []string
+
+	topoType := ""
+	if bnc.IsSecondary() {
+		topoType = bnc.GetTopologyType()
 	}
 
-	var ops []ovsdb.Operation
-	for _, n := range nodes {
-		// skip nodes that are not running ovnk (inferred from host subnets)
-		switchName := n.Name
-		if bnc.lsManager.IsNonHostSubnetSwitch(switchName) {
-			continue
+	if !bnc.IsSecondary() || topoType == ovntypes.Layer3AttachDefTopoType {
```

Comment:
for default network, we cannot call  GetTopologyType(). Probably I would just implement the netconfInfo for default network as well.

Reply:
Maybe here is just that `deleteStaleLogicalSwitchPorts` should be implemented per controller instead of in the `BaseNetworkController`. That implementation could use a `GetExpectedPodSwitchName(node)` that could be different depending on the topology type and then call the generic `deleteStaleLogicalSwitchPortsOnSwitches`

Reply:
in this particular case, the switchName is bnc.GetPrefix()+nodeName for default network and layer3 networks. but for upcoming layer2/localnet networks, switchName would be different. I can reorganize the code to make it clear.